connector/src/main/java/org/geysermc/connector/health/GeyserHealthProvider.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2019-2021 GeyserMC. http://geysermc.org
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.connector.health;
+
+import org.geysermc.connector.GeyserLogger;
+import org.geysermc.connector.configuration.GeyserConfiguration;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class GeyserHealthProvider extends Thread {
+
+    private final GeyserConfiguration.IHealthConfiguration healthConfiguration;
+    private final GeyserLogger logger;
+
+    public GeyserHealthProvider(GeyserConfiguration.IHealthConfiguration healthConfiguration, GeyserLogger logger) {
+        this.healthConfiguration = healthConfiguration;
+        this.logger = logger;
+    }
+
+    @SuppressWarnings("InfiniteLoopStatement")
```

Comment:
Yeah, it's a warning which will be there if we just remove the suppression, the suppression is just to recognize "we know it is that, but we don't care".

Reply:
I was referring to the while loop, sorry I wasn't clear.

Reply:
Ah yeah, unfortunately I don't think there's a better way to force an infinite loop here. It's in its own thread so it shouldn't actually cause problems. It's also set as a daemon so it will always be reliant on the main thread being running. It should properly look at itself as a service so the infinite loop won't actually cause any issues. It's a codesmell sometimes but I think here it's warranted.